src/components/newsletter-form.js

```diff
@@ -54,52 +69,31 @@ export function useNewsletterForm() {
 
       setLoading(true);
 
-      const programResp = await fetch(`${wwwFunctionsRootPath}/program`, {
-        method: 'POST',
-        headers: {
-          Accept: 'application/json',
-          'Content-Type': 'application/json'
-        },
-        body: JSON.stringify({
-          id: 1172
-        })
-      });
-
-      const programName = await programResp.json();
-
-      const obj = {
-        programName,
-        source: 'Email Signup',
-        lookupField: 'email',
-        input: [
-          {
-            email,
-            Lead_Source_Most_Recent__c: 'Email Signup',
-            Lead_Source_Detail__c: 'Email Signup Blog',
-            UTM_Medium__c: utmMedium,
-            UTM_Source__c: utmSource,
-            UTM_Campaign__c: utmCampaign,
-            UTM_Term__c: utmTerm
+      try {
+        pushMarketoLead({
+          variables: {
+            source: 'Email Signup',
+            programId: '1172',
+            input: {
+              email,
+              Lead_Source_Most_Recent__c: 'Email Signup',
+              Lead_Source_Detail__c: 'Email Signup Blog',
+              UTM_Medium__c: utmMedium,
+              UTM_Source__c: utmSource,
+              UTM_Campaign__c: utmCampaign,
+              UTM_Term__c: utmTerm
+            }
           }
-        ]
-      };
-
-      const response = await fetch(`${wwwFunctionsRootPath}/marketo`, {
-        method: 'POST',
-        headers: {
-          Accept: 'application/json',
-          'Content-Type': 'application/json'
-        },
-        body: JSON.stringify(obj)
-      });
-
-      if (response.ok) {
-        setSuccess(true);
-        window.gtag?.('event', 'blog_newsletter_form_submit');
-        return;
+        });
+      } catch (err) {
+        console.log('marketing graph error');
+        setLoading(false);
+        return false;
```

Comment:
Are we tracking these errors anywhere?

Reply:
@BlenderDude we're handling the majority of errors in the graph with the zapier workflow where the leads and request object are pushed into that google sheet - but i'm realizing now there's a couple places where we just have console logs in place in the event of an error, i'll get a pr up later this week/early next week for you and @trevorblades to take a peek at that will make sure we're capturing all of the errors.

thanks for the callout!

marketing graph for reference:

https://github.com/apollographql/marketing/blob/main/netlify/functions/graphql/schema.ts